File: src/Cell.cpp
#include <iostream>
#include <string>
#include "../include/Cell.h"
using namespace std;

/*Constructor untuk map (cell)*/
Cell::Cell() {
    sizex = 15;
    sizey = 15;
    map = new char**[sizex];
    for(int i = 0; i <sizex; i++) {
        map[i] = new char*[sizey];
        for(int j = 0; j < sizey; j++) {
            map[i][j] = new char[3];
                map[i][j][0] = '-';
                map[i][j][1] = ' ';
                map[i][j][2] = ' ';
        }
    }
    for(int i = 0; i < 5; i++) {
        for(int j = 0; j < 5; j++) {
            map[i][j][0] = 'o';
            map[i][j][1] = '*';
            map[i][j][2] = ' ';
        }
    }
    for(int i = 6; i < 11; i++) {
        for(int j = 0; j < 7; j++) {
            map[i][j][0] = 'x';
            map[i][j][1] = '@';
            map[i][j][2] = ' ';
        }
    }
}
/*Dtor untuk map (cell)*/
Cell::~Cell() {
    for(int i = 0; i < sizex; i++) {
        for(int j = 0; j < sizey; j++) {
            delete[] map[i][j];
        }
        delete[] map[i];
    }
    delete[] map;
}

/*Mengubah nilai sizex menjadi _x*/
void Cell::setSizex(int _x) {
    sizex = _x;
}
/*Mengubah nilai sizey menjadi _y*/
void Cell::setSizey(int _y) {
    sizey = _y;
}
/*Mengembalikan sizex map*/
int Cell::getSizex() {
    return sizex;
}
/*Mengembalikan sizex map*/
int Cell::getSizey() {
    return sizey;
}
/*Mengembalikan value dari matriks map*/
char Cell::getElement(int x, int y) {
    return map[x][y][2];
}
/*Mengembalikan jenis land sesuai dengan value dari matriks map*/
char Cell::getTypeLand(int x, int y){
    return map[x][y][0];
}
/*Mengecek cell terisi atau tidak*/
bool Cell::isEmptyCell(int x, int y) {
    if(map[x][y][2] != ' ') {
        return true;
    }else {
        return false;
    }
}

/*Menentukan nilai value dari cell pada matriks*/
void Cell::setElement(int x, int y, char e) {
    map[x][y][2] = e;
}

/*
*Menentukan petak rumput sesuai dengan land
*/
void Cell::setRumput(int x, int y){
    if(map[x][y][0] == '-'){
        map[x][y][1] = '#';
    }else if(map[x][y][0] == 'o'){
        map[x][y][1] = '*';
    }else{
        map[x][y][1] = '@';
    }
}

/*Mengembalikan nilai ada rumput atau tidak*/
char Cell::getRumput(int x, int y){
    return map[x][y][1];
}

File: java/src/Board/Cell.java
/**
 * Cell Class.
 */
class Cell {
    /** The X size of Board */
    int sizex; 
    /** The Y size of Board */
    int sizey; 

    /** Map Matrix */
    char[][][] map;

    /**
     * Ctor
     */
    public Cell() {
        sizex = 15;
        sizey = 15;
        map = new char[sizex][sizey][3];
        for (int i = 0; i <sizex; i++) {
            for (int j = 0; j < sizey; j++) {
                    map[i][j][0] = '-';
                    map[i][j][1] = '#';
                    map[i][j][2] = ' ';
            }
        }
        for (int i = 0; i < 5; i++) {
            for (int j = 0; j < 5; j++) {
                map[i][j][0] = 'o';
                map[i][j][1] = '*';
                map[i][j][2] = ' ';
            }
        }
        for (int i = 6; i < 11; i++) {
            for (int j = 0; j < 7; j++) {
                map[i][j][0] = 'x';
                map[i][j][1] = '@';
                map[i][j][2] = ' ';
            }
        }
    }

    /**
     * Change the value of X size.
     *
     * @param _x The new X size.
     */
    public void setSizeX(int _x) {
        sizex = _x;
    }

    /**
     * Change the value of Y size.
     *
     * @param _y The new Y size.
     */
    public void setSizeY(int _y) {
        sizey = _y;
    }

    /**
     * X size getter.
     *
     * @return int The X size.
     */
    public int getSizeX() {
        return sizex;
    }

    /**
     * Y size getter.
     *
     * @return int The Y size.
     */
    public int getSizeY() {
        return sizey;
    }

    /**
     * Map matrix value getter.
     *
     * @param x The x position.
     * @param y The y position.
     * @return int The map matrix value.
     */
    public char getElement(int x, int y) {
        return map[x][y][2];
    }

    /**
     * Map matrix land type getter.
     *
     * @param x The x position.
     * @param y The y position.
     * @return int The map matrix land type.
     */
    public char getTypeLand(int x, int y) {
        return map[x][y][0];
    }

    /**
     * Check whether the cell is empty or not.
     *
     * @param x The x position.
     * @param y The y position.
     * @return bool True if filled, otherwise False.
     */
    public boolean isEmptyCell(int x, int y) {
        if (this.getElement(x, y) == ' ') {
            return true;
        } else if ((this.getElement(x, y) == 'P') ||
            (this.getElement(x, y) == 'M') ||
            (this.getElement(x, y) == 'W') ||
            (this.getElement(x, y) == 'T')) {
            return false;
        } else {
            return false;
        }
    }

    /**
     * Map matrix value setter.
     *
     * @param x The x position.
     * @param y The y position.
     * @param e The value to be inserted.
     */
    public void setElement(int x, int y, char e) {
        map[x][y][2] = e;
    }

    /**
     * Map matrix grass setter according to land type.
     *
     * @param x The x position.
     * @param y The y position.
     */
    public void setRumput(int x, int y) {
        if(map[x][y][0] == '-') {
            map[x][y][1] = '#';
        } else if(map[x][y][0] == 'o') {
            map[x][y][1] = '*';
        } else{
            map[x][y][1] = '@';
        }
    }

    /**
     * Map matrix empty grass setter.
     *
     * @param x The x position.
     * @param y The y position.
     */
    public void kosongRumput(int x,int y) {
        map[x][y][1] = ' ';
    }

    /**
     * Check whether there is grass.
     *
     * @param x The x position.
     * @param y The y position.
     * @return bool True if exist grass on position, otherwise False..
     */
    public char getRumput(int x, int y) {
        return map[x][y][1];
    }
}